### ISSUE TITLE
Investigation of current branches of this project.

### ISSUE THREAD

**Issue by denisfa** [visualboyadvance-m/visualboyadvance-m#497]
I will keep some logs and comments about each branch, as well as the dates of deletion.
Started on `01/09/2019` (first of september, 2019).

- [ ] `agbprint`
- [ ] `arthur/PNaCl`
- [ ] `arthur/dsound`
- [ ] `arthur/gles`
- [ ] `arthur/master`
- [ ] `arthur/opengl-ifdef`
- [ ] `arthur/sdl2`
- [ ] `arthur/sergiobenrocha2-master`
- [ ] `arthur/svn-master`
- [ ] `arthur/svn_adamn-link`
- [ ] `arthur/svn_bgk-link`
- [ ] `arthur/wxV3`
- [ ] `debian-update`
- [ ] `linux-static-build`
- [ ] `no-debugger`
- [ ] `osx-32bit`
- [ ] `utf8-logger`
- [ ] `vbam-rr`

**Comment by ZachBacon** [comment]
I think the linux-static-build should stay as I think @rkitover still has some plans for that. I'm not 100% on that yet.

**Comment by denisfa** [comment]
@ZachBacon No issues, this is mostly an investigation. If it is indeed valid work, I will try to merge it, just like other branch dedicated to a specific issue.

**Comment by ZachBacon** [comment]
@denisfa I think @rkitover would prefer not to merge that in just yet, WIP stuff and all.

**Comment by ZachBacon** [comment]
outright though I think the arthur branches for SDL2 and wxv3 probably could be removed since we either already migrated on our own to SDL2 and his changes are primarily unfinished work there.

**Comment by denisfa** [comment]
@ZachBacon Sorry, I meant finish whatever WIPs are there and then make a PR. I did this to `xbrz1.6` (updated to newer version etc) and some other.

**Comment by ZachBacon** [comment]
I understand, but I'm saying @rkitover's linux-static branch is more of a personal project I think that uses his build script, and the other two branches I mentioned we tackled on and technically completed in master.

**Comment by denisfa** [comment]
@ZachBacon Ok, I will take a look and check these.

**Comment by rkitover** [comment]
deleted `osx-32bit` as that was already merged

**Comment by rkitover** [comment]
Deleted `agbprint`.

**Comment by rkitover** [comment]
@denisfa thanks for spearheading this effort, all of our branches have been cleaned up now.

The only thing left is to look at the arthur branches and see if there is any useful code there.

**Comment by denisfa** [comment]
@rkitover @ZachBacon What are these tags:
```
Beta-3 # THIS ONE
Keyboardfixes # THIS ONE
VBA-M_Beta_2 # THIS ONE
c7c6ad6 # THIS ONE
throttle # THIS ONE
v2.0.0
v2.0.1
v2.0.2
v2.1.0
v2.1.1
v2.1.2
v2.1.3
waylandplus # THIS ONE
```

Could we create a task to remove some of these too? I know they only tag commits, but is there a reason for all of them to be there?

**Comment by ZachBacon** [comment]
Old test releases before we started doing them differently, Keep them up I think for now, But I'll leave that call to @rkitover

**Comment by rkitover** [comment]
The tag code in cmake ignores these, so they do no harm. You can delete them if you want though.

**Comment by rkitover** [comment]
@denisfa We can rewrite those tags as version numbers with the strings as the tag messages, sound good? I can do that.

**Comment by denisfa** [comment]
> We can rewrite those tags as version numbers with the strings as the tag messages, sound good? I can do that.

@rkitover Yes, please. They are just weird to look at. There is the continuous versioning and the others. When I saw those, it felt like `Keyboardfixes`, `c7c6ad6`, `waylandplus` and `throttle` were leftovers from something from the past (like checking a solution for an issue). Meanwhile, `VBA-M_Beta_2` and `Beta-3` could be `v1.X.X` or something like that.

Also, what happened to the SVN repos? Are they lost forever? It would be really nice if we could check previous code, especially for some of the `link` issues.

**Comment by rkitover** [comment]
@ZachBacon would know more about the svn repos, may be what was on the sourceforge project here:

https://sourceforge.net/projects/vbam/

**Comment by ZachBacon** [comment]
those in the svn repo should of been imported as branches here on git when I did the migration.

**Comment by denisfa** [comment]
@rkitover @ZachBacon The branch `arthur/PNaCl` was supposed to be make the SDL port run as a web app, but its main lib has been deprecated by google (sources [1](https://blog.chromium.org/2017/05/goodbye-pnacl-hello-webassembly.html) and [2](https://groups.google.com/forum/#!topic/native-client-discuss/oMHnO-6RAas)). There is an alternative called [webassembly](https://webassembly.org/), and also a [migration plan](https://developer.chrome.com/native-client/migration).

Do you think it is worth to go after this? The current branch does not compile, since it is incomplete and it uses a very old version of the `PNaCl` lib. Personally, I think it could be a good learning experience, but I would have to start from 0. For now, I will keep this branch since that is the main part of all this.

One thing to notice is that all these branches (`gles`, `dsound`, `PNaCl`, `sdl2`, `opengl-ifdef`) were all fragmented work to prepare the SDL port to run both web and the binary we run today. None of them were completed (`dsound` for example was never started, but there were commits fixing other stuff that were also commited to the current SDL code, same as `gles` and `sdl2`).

With that said, the `arthur/master` was the origin of all these. So we can delete it after we are done here.

**Comment by denisfa** [comment]
@ZachBacon Check [this](https://github.com/visualboyadvance-m/visualboyadvance-m/commit/05435d0f342ee23307154403b7adb55f87b1720a). What does

```git-svn-id: https://svn.code.sf.net/p/vbam/code/branches/bgk-link@1230 a31d4220-a93d-0410-bf67-fe4944624d44```

mean? Is this recoverable?

**Comment by ZachBacon** [comment]
That's autogenerated from when I pulled the svn repo into github.

**Comment by ZachBacon** [comment]
As for the pncl stuff and the gles, dsound, sdl2 etc they weren't completed works, Most of his efforts went into the frame stuff and making frames accessible from all frontends at the time.

**Comment by rkitover** [comment]
@denisfa Here you go, a bit of revisionist history, nice and clean:

```bash
git tag -l --sort='*authordate' --format='%(refname:short)   %(contents:subject)   %(*authordate:short)   %(*objectname:short)'
v1.9.4   v1.9.4   2015-09-19   abe2c0f2
v1.9.5   v1.9.5   2016-08-09   1eb76857
v1.9.6   v1.9.6   2016-12-14   f7f67ff6
v1.9.7   v1.9.7   2017-02-02   6ecab805
v1.9.8   v1.9.8   2017-02-22   931fda45
v1.9.9   v1.9.9   2017-03-14   2e8bee11
v2.0.1   v2.0.1   2017-10-10   55fa9c69
v2.0.2   v2.0.2   2018-03-13   92ba6766
v2.1.0   v2.1.0   2018-06-30   7aa5d939
v2.1.1   v2.1.1   2019-02-25   b5741ee4
v2.1.2   v2.1.2   2019-04-06   95433f6c
v2.1.3   v2.1.3   2019-04-17   1401573f
```

**Comment by denisfa** [comment]
@rkitover This is amazing! Thank you!

**Comment by Dune-jr** [comment]
This looks old, should it be closed? perhaps with git [branch descriptions](https://git-scm.com/docs/git-branch#_description)?
